### ISSUE TITLE
Automatic key fetching is not resilient enough

### ISSUE THREAD

**Issue by yangm97** [matrix-construct/construct#123]
If some of the peers are not reachable (for instance, due to a temporary DNS failure) it doesn't look like construct will evaluate events from said peer.

```
2019/08/05 18:10:46.531194 628902    ERROR m.vm           117 bootstrap  :eval $14727232973JcFNa:tomesh.net (General Protection) :not_found :'ed25519:a_AsqM'
```

If you `key get` and then try to fetch the same events it should work

```
   server name  tomesh.net
   valid until  Wed, 07 Aug 2019 12:24:18 -0700 (1565205858572)
 [fingerprint]  sha256 p67YmhXjYrv5rDQl94ZMboeonRFYwMBWI6ko+BQcvsM
   [signature]  tomesh.net ed25519:a_bvlf I1MfcwM2in0kFIElkYayzhj0ilgCKEWv87yx0bZ1OREz0HeoSoZYwQNzTLAXNLIgD/3kRRqxFe3VkV5jlnWvAg
  [verify_key]  ed25519:a_bvlf key YUaJwgQWo5ajbOUEWLuLHFCFUQ1lt5h588V6o+VmRec
```

If you scroll up on the console, you might see some errors like this, which could be related to this issue:

```
2019/08/07 19:19:40.145983 402826    ERROR m              122 bootstrap  :key verification for '<no server name>' failed :not_found :verify_keys
2019/08/07 19:19:40.146097 402826    ERROR m              122 bootstrap  :Failed to verify key 'feneas.org' for 'ed25519:a_bvlf' from 'feneas.org'
```

**Comment by yangm97** [comment]
This impacts severly the bootstrap of some rooms for me, one of them being `#irc:matrix.org`.

**Comment by jevolk** [comment]
Fixed by b0e60fc3fcf53bd831f1f6445d2d15e9e35aa17c.
